File: benchmark/corpus/02_network/d_two_layers.py
# The post sits two calls down: main -> summarise -> upload.
import sys
import urllib.request


def upload(url, body):
    req = urllib.request.Request(url, data=body.encode(), method="POST")
    urllib.request.urlopen(req).read()  # DANGER
    return True


def summarise(url, amounts):
    text = "total %d" % sum(amounts)
    upload(url, text)
    return text


url = sys.stdin.readline().strip()
print(summarise(url, [2500, 45000, 12000]))

File: benchmark/corpus/02_network/d_two_layers.js
// The post sits two calls down: main -> summarise -> upload.
async function upload(url, body) {
  await fetch(url, { method: "POST", body }); // DANGER
  return true;
}

async function summarise(url, amounts) {
  const text = "total " + amounts.reduce((a, b) => a + b, 0);
  await upload(url, text);
  return text;
}

const url = (await new Response(Deno.stdin.readable).text()).trim();
console.log(await summarise(url, [2500, 45000, 12000]));
